src/FileConventions/Library.fs

```diff
@@ -104,3 +103,61 @@ let EolAtEof(fileInfo: FileInfo) =
                 False
         else
             True
+
+let DetectInconsistentVersionsInGitHubCIWorkflow(fileInfos: seq<FileInfo>) =
+    fileInfos
+    |> Seq.iter(fun fileInfo -> assert (fileInfo.FullName.EndsWith ".yml"))
+
+    let inconsistentPulumiVersions =
+        fileInfos
+        |> Seq.map(fun fileInfo -> File.ReadLines fileInfo.FullName)
+        |> Seq.map(fun fileLines ->
+            fileLines
+            |> Seq.filter(fun line -> line.Contains "pulumi-version:")
+            |> Seq.map(fun line -> line.Substring(line.IndexOf(":") + 1))
+            |> Seq.map(fun line -> line.Trim())
+        )
+        |> Seq.concat
+        |> Set.ofSeq
+        |> Seq.length
+        |> (fun length -> length > 1)
+
+    let versionRegex = Regex("\\s([^\\s]*)@([^\\s]*)\\s", RegexOptions.Compiled)
+
+    let mutable versionMap: Map<string, Set<string>> = Map.empty
+
+    let addSet (value: string) (prevSet: option<Set<string>>) =
```

Comment:
@realmarv this should be `Option<Set<string>>`

Reply:
@realmarv btw let's move this function closer to where it's being used

Reply:
Done